GRDB/Record/FetchableRecord.swift

```diff
@@ -592,3 +610,80 @@ public enum DatabaseDateDecodingStrategy {
     /// error, and react accordingly).
     case custom((DatabaseValue) -> Date?)
 }
+
+// MARK: - DatabaseKeyDecodingStrategy
+
+/// `DatabaseKeyDecodingStrategy` specifies how `FetchableRecord` types that
+/// also adopt the standard `Decodable` protocol decode their keys.
+///
+/// For example:
+///
+///     struct Player: FetchableRecord, Decodable {
+///         static let databaseKeyDecodingStrategy = DatabaseKeyDecodingStrategy.convertFromSnakeCase
+///
+///         var playerID: Int // decoded from player_id
+///     }
+public enum DatabaseKeyDecodingStrategy {
+    /// Uses the original key without applying any transformations
+    case useDefaultKeys
+    /// Converts the key from snake case format, e.g. player_id to playerID
+    case convertFromSnakeCase
+
+    func convertedKey(key: String) -> String {
+        switch self {
+        case .useDefaultKeys:
+            return key
+        case .convertFromSnakeCase:
+            return Self.convertToSnakeCase(key)
+        }
+    }
+
+    fileprivate static func convertToSnakeCase(_ stringKey: String) -> String {
```

Comment:
This implementation is copied verbatim from https://github.com/apple/swift-corelibs-foundation/blob/eec4b26deee34edb7664ddd9c1222492a399d122/Sources/Foundation/JSONEncoder.swift#L127-L175

Reply:
Good idea 👍 

We should copy the [license](https://github.com/apple/swift-corelibs-foundation/blob/eec4b26deee34edb7664ddd9c1222492a399d122/Sources/Foundation/JSONEncoder.swift#L5-L9) somewhere, maybe inside the function body.